packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -116,4 +128,68 @@ contract L1Block is ISemver {
             sstore(batcherHash.slot, calldataload(132)) // bytes32
         }
     }
+
+    /// @notice Updates the L1 block values for an interop upgraded chain.
+    /// @dev The interop upgrade block itself MUST include a call to setL1BlockValuesEcotone.
+    /// Params are packed and passed in as raw msg.data instead of ABI to reduce calldata size.
+    /// Params are expected to be in the following order:
+    ///   1. _baseFeeScalar      L1 base fee scalar
+    ///   2. _blobBaseFeeScalar  L1 blob base fee scalar
+    ///   3. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   4. _timestamp          L1 timestamp.
+    ///   5. _number             L1 blocknumber.
+    ///   6. _basefee            L1 base fee.
+    ///   7. _blobBaseFee        L1 blob base fee.
+    ///   8. _hash               L1 blockhash.
+    ///   9. _batcherHash        Versioned hash to authenticate batcher by.
+    ///  10. _interopSetSize     Size of the interop dependency set.
+    ///  11. _chainIds           Array of chain IDs in the interop dependency set.
+    function setL1BlockValuesInterop() external {
+        assembly {
+            // Revert if the caller is not the depositor account.
+            if xor(caller(), DEPOSITOR_ACCOUNT) {
+                mstore(0x00, 0x3cc50b45) // 0x3cc50b45 is the 4-byte selector of "NotDepositor()"
+                revert(0x1C, 0x04) // returns the stored 4-byte selector from above
+            }
+
```

Comment:
This function also needs to set the other values in storage besides just the dependency set

Reply:
I don't follow exactly - the idea is that the first tx in every block is always the L1 attributes tx, we cannot break that invariant. The upgrade txs will deploy + upgrade the L1Block predeploy afterwards meaning that the interop interface isn't available to be called until interop activation height + 1

The smart contract parses the calldata built in Go when creating the L1 attributes tx, it still passes the rest of the values from the ecotone upgrade, the interop upgrade extends the ecotone upgrade